### PR TITLE
Cross Browser Compatibility Testing AND retrieving and rendering Thread metrics

### LINKED ISSUE #32: Implement ability to view thread information and stacks
This should be functionally similar to the Twitter Server treads view:
image

This information does not seem to come from `metrics.json'.

### LINKED ISSUE #59: Fix chart rendering on Safari and IE11

### PR DESCRIPTION
**Pull request by spmcbride1201**
Resolve #32 
Resolve #59

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -13,14 +13,35 @@ const metrics = State({
...
+        let threadIds = Object.keys(value);
+        const results = threadIds.map(id => ({
+          name: value[id].thread,
+          id: id,
+          priority: value[id].priority,
+          state: value[id].state,
+          daemon: value[id].daemon,
+          stack: value[id].stack
+        })
+        );
```
Comment (tilleryd) [inline src/store.js]: Can you move this closing parenthesis to line 28?

File: src/style/components/jsonInspector.less [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// components/navbar.less
```
Comment (tilleryd) [inline src/style/components/jsonInspector.less]: components/jsonInspector.less

File: src/style/components/table.less [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// components/navbar.less
```
Comment (tilleryd) [inline src/style/components/table.less]: components/table.less

File: src/style/components/jsonInspector.less (hunk 2 above)
Comment (tilleryd) [inline src/style/components/jsonInspector.less]: was the css from this file copied from jsonInspector?  if so, can we not import via Webpack?

File: src/utils.js [hunk 4]
```diff
@@ -1,25 +1,59 @@
...
 
+// Dashboard Utility Functions
+
+// The client-side Redux store expresses metrics over time as a hierarchy of JavaScript objects. 
+// At the lowest level, a metric is represented by a key of the metric name and a complex object
+// representing the value of that metric over time. The complex object has keys 
+// equal to a UNIX timestamp and values equal to the value of the metric at the associated timestamp.
+// When the client fetches metrics data from the server, the data is deconstructed and the value
+// of each attribute is appended to the appropriate complex object with a key equal to the timestamp
+// of when the fetched data was received from the server.
```
Comment (tilleryd) [inline src/utils.js]: 👍 Nice explanation

Comment (tilleryd) [inline src/utils.js]: I really like the detailed code comments for all of these utility functions.

File: src/components/JVMThreadsTable.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+              <th className="uk-text-truncate">State</th>
+              <th className="uk-text-truncate">Daemon</th>
+              <th className="uk-text-truncate">Priority</th>
+            </tr>
+          </thead>
+          <tbody>
+            {threadsTable.map(({ id, name, priority, state, daemon, stack }) => {
+              return (
+                <tr
+                  data-uk-="pos: bottom-left"
```
Comment (spmcbride1201) [inline src/components/JVMThreadsTable.js]: lol, there's no way it is! ty!

File: src/style/components/jsonInspector.less (hunk 2 above)
Comment (spmcbride1201) [inline src/style/components/jsonInspector.less]: For some reason, the npm module didn't seem to contain the css file. I had to pull it from their GitHub source.

File: src/utils.js (hunk 4 above)
Comment (spmcbride1201) [inline src/utils.js]: The really cool thing is that VSCode displays the comments about the function and the signature of the params when you hover over any use of that function throughout the project. I'm sort of equating these utility functions with what would normally be an API, so I'm trying to document the surface area, assuming that the main thing that people would contribute would be custom charts.

**Pull request merged**